### PR TITLE
Send credentials if needed before kyuubi operation run to prevent token expired issue

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

We met token expired issue.

```
"org.apache.kyuubi.KyuubiSQLException: org.apache.kyuubi.KyuubiSQLException: Error operating ExecuteStatement: org.apache.hadoop.ipc.RemoteException(org.apache.hadoop.security.token.SecretManager$InvalidToken): token (HDFS_DELEGATION_TOKEN token 57938358 for b_stf) can't be found in cache
at org.apache.hadoop.ipc.Client.getRpcResponse(Client.java:1907)
at org.apache.hadoop.ipc.Client.call(Client.java:1831)
at org.apache.hadoop.ipc.Client.call(Client.java:1719)
at org.apache.hadoop.ipc.Client.call(Client.java:1648)
at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:260)
at com.sun.proxy.$Proxy16.getFileInfo(Unknown Source)
at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getFileInfo(ClientNamenodeProtocolTranslatorPB.java:1184)
at sun.reflect.GeneratedMethodAccessor44.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:306)
at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:207)
at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:189)
at com.sun.proxy.$Proxy17.getFileInfo(Unknown Source)
at org.apache.hadoop.hdfs.DFSClient.getFileInfo(DFSClient.java:2184)
```

The  engine share level is CONNECTION.

We keep the connection alive with KyuubiConnection::isValid, the max idle time is 3d.

There is no statements executed in last 24h and it  caused the hdfs delegation expired.

Then I want to execute a statement and met above issue.

### _How was this patch tested?_
Pass existing UT.

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala [hunk 1]
```diff
@@ -59,6 +59,7 @@ class ExecuteStatement(
 
   private def executeStatement(): Unit = {
     try {
+      sendCredentialsIfNeeded()
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala:62]: what happens if call other operations ? e.g. GetTables

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala:62]: these operations also might meet token expiration issue.

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala:62]: should we do it in KyuubiOperation::beforeRun?
For other operations, most of them rely on hive token.
cc @yaooqinn

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala:62]: seems hive token lift time is more long than hdfs delegation token, so, it is not very critical for us.

**Comment by turboFei** [comment]
cc @ulysses-you @yaooqinn @cxzl25

**Comment by turboFei** [comment]
Also cc @zhouyifan279

**Review by cxzl25 — APPROVED** [review]

**Review by ulysses-you — APPROVED** [review]

**Comment by turboFei** [comment]
thanks @ulysses-you @pan3793  @cxzl25 , merging to master/branch-1.6

**Pull request closed without merging**